pkg/syncer/cnsoperator/util/util.go

```diff
@@ -209,3 +215,86 @@ func GetNetworkProvider(ctx context.Context) (string, error) {
 	return "", fmt.Errorf("could not find network provider field in configmap %q in namespace %q",
 		wcpNetworkConfigMap, kubeSystemNamespace)
 }
+
+// GetVCDatacenterFromConfig returns datacenter registered for each vCenter
+func GetVCDatacentersFromConfig(cfg *config.Config) ([]string, string, error) {
+	dcList := make([]string, 0)
+	vcHost := ""
+	if len(cfg.VirtualCenter) > 1 {
+		return dcList, vcHost, fmt.Errorf("invalid configuration. Expected only 1 VC but found %d", len(cfg.VirtualCenter))
+	}
+
+	for host, value := range cfg.VirtualCenter {
+		vcHost = host
+		datacentersOnVcenter := strings.Split(value.Datacenters, ",")
+		for _, dc := range datacentersOnVcenter {
+			dcMoID := strings.TrimSpace(dc)
+			if dcMoID != "" {
+				dcList = append(dcList, dcMoID)
+			}
+		}
+	}
+	if len(dcList) == 0 {
+		return dcList, vcHost, errors.New("unable get vCenter datacenters from vsphere config")
+	}
+	return dcList, vcHost, nil
+}
+
+// GetVMFromVcenter returns the VM from vCenter for the given nodeUUID.
+func GetVMFromVcenter(ctx context.Context, nodeUUID string,
+	configInfo config.ConfigurationInfo) (*cnsvsphere.VirtualMachine, error) {
+	log := logger.GetLogger(ctx)
+
+	dcList, err := GetDatacenterObjectList(ctx, configInfo)
+	if err != nil {
+		log.Errorf("failed to get datacenter for node: %s. Err: %+q", nodeUUID, err)
+		return nil, err
+	}
+
+	for _, dc := range dcList {
+		vm, err := dc.GetVirtualMachineByUUID(ctx, nodeUUID, false)
```

Comment:
`GetVirtualMachineByUUID` uses vm instance uuid if the parameter passed os `true`, please check that and update it.